### PR TITLE
Decoration extension

### PR DESCRIPTION
**Pull request by Elorm-Axolu**
# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

**Review by nwokafor-choongsaeng — CHANGES_REQUESTED** [review]
A question I couldn't find an answer while looking at the changes is "what does the extension do when you load up a morphir project that doesn't have decorations configured"?

File: cli/web/valueEditor.html
[the remark this answers is not in the document]
Comment (Elorm-Axolu) [reply cli/web/valueEditor.html:1]: I created this to test the editor exposed

File: morphir-decoration-extension/src/editor.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+): vscode.WebviewOptions {
+  return {
+    // Enable javascript in the webview
+    enableScripts: true,
+
+    // And restrict the webview to only loading content from our extension's `media` directory.
+    localResourceRoots: [
+      vscode.Uri.joinPath(extensionUri, "media"),
+      vscode.Uri.joinPath(extensionUri, "../cli/web"),
+      vscode.Uri.joinPath(extensionUri, "../test-integration"),
```
Comment (AttilaMihaly) [inline morphir-decoration-extension/src/editor.ts:16]: Are all these paths required? The comment only mentions the first one.

File: morphir-decoration-extension/src/getNonce.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export function getNonce() {
```
Comment (AttilaMihaly) [inline morphir-decoration-extension/src/getNonce.ts:1]: What's the purpose of this? Can you add some comments?

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]
Looks good. I added a few questions and change requests.

File: morphir-decoration-extension/src/editor.ts (hunk 1 above)
Comment (Elorm-Axolu) [reply morphir-decoration-extension/src/editor.ts:16]: restrict the webview to only loading content from the directory paths added in the localResourcesRoots

File: morphir-decoration-extension/src/getNonce.ts (hunk 2 above)
Comment (Elorm-Axolu) [reply morphir-decoration-extension/src/getNonce.ts:1]: I forgot to clean that up, It is currently not being used

**Review by AttilaMihaly — APPROVED** [review]

**Pull request merged**